lib/cct/commands/openstack.rb

```diff
@@ -69,15 +74,29 @@ def create name, options={}
           OpenStruct.new(shell_parse(exec!(all_params).output))
         end
 
+        def add name, options={}
+          params.clear
+          yield params
+          all_params = ["add", name, "--format=shell"].concat(params.extract!(options))
+          OpenStruct.new(shell_parse(exec!(all_params).output))
+        end
+
+        def set name, options={}
+          params.clear
+          yield params
+          all_params = ["set", name].concat(params.extract!(options))
+          OpenStruct.new(shell_parse(exec!(all_params).output))
+        end
+
         def delete id_or_name
           params.clear
           exec!("delete", id_or_name)
         end
 
-        def list
+        def list *options
           params.clear
-          user_row = Struct.new(:id, :name)
-          result = exec!("list", "--format=csv").output
+          user_row = Struct.new(:id, :name, :subnets)
```

Comment:
I would rather go with checking the last item of `options` for hash and search for `:row` param which might provide the exact struct as needed.  Something like:
```ruby
params = options.last.is_a?(Hash) ? options.last : {}
row = params[:row] || Struct.new(:id, :name)
```
It's better to keep the parent class as much generic as possible.